temporal-serviceclient/src/main/java/io/temporal/serviceclient/WorkflowServiceStubsImpl.java

```diff
@@ -138,7 +140,7 @@ public WorkflowServiceStubsImpl(
         new GrpcMetricsInterceptor(options.getMetricsScope());
     ClientInterceptor deadlineInterceptor = new GrpcDeadlineInterceptor(options);
     GrpcTracingInterceptor tracingInterceptor = new GrpcTracingInterceptor();
-    Metadata headers = new Metadata();
+    Metadata headers = options.getHeaders();
```

Comment:
I would merge options headers here. Mutating ones from the options is not clean.

Reply:
Done.